### PR TITLE
[15.0][MIG] maintenance_project: Migration to 15.0

### PR DESCRIPTION
**Pull request by mymage**

### CODE REVIEW

**Comment by dalonsod** [comment]
Hello @mymage thanks for the contribution!

Inspecting tests error, maybe demo data references should be replaced by simply local variables created during tests. AFAIK using demo data in tests is not recommended (-1 for me, as I was the original author)

**Comment by mymage** [comment]
> Hello @mymage thanks for the contribution!
> 
> Inspecting tests error, maybe demo data references should be replaced by simply local variables created during tests. AFAIK using demo data in tests is not recommended (-1 for me, as I was the original author)

Hi @dalonsod, I didn't forget this and now I'm updating my todo list, hope to fix it soon.

**Review by dreispt — APPROVED** [review]

File: maintenance_project/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2019 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+{
+    "name": "Maintenance Projects",
+    "summary": "Adds projects to maintenance equipments and requests",
+    "version": "15.0.1.1.0",
```
Comment (victoralmau) [inline maintenance_project/__manifest__.py]:
```suggestion
    "version": "15.0.1.0.0",
```

File: maintenance_project/data/demo_maintenance_project.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<odoo noupdate="1">
+    <record id="project_project_1" model="project.project">
+        <field name="name">General equipment project</field>
+        <field name="user_id" ref="base.user_admin" />
+    </record>
+    <record id="project_task_11" model="project.task">
+        <field name="user_ids" ref="base.user_admin" />
```
Comment (victoralmau) [inline maintenance_project/data/demo_maintenance_project.xml]:
```suggestion
        <field name="user_ids" eval="[(4, ref('base.user_admin'))]" />
```

File: maintenance_project/data/demo_maintenance_project.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        <field name="project_id" ref="maintenance_project.project_project_1" />
+        <field name="name">General task</field>
+    </record>
+    <record id="project_task_12" model="project.task">
+        <field name="user_ids" ref="base.user_admin" />
```
Comment (victoralmau) [inline maintenance_project/data/demo_maintenance_project.xml]:
```suggestion
        <field name="user_ids" eval="[(4, ref('base.user_admin'))]" />
```

File: maintenance_project/models/maintenance_request.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        if not newreq.project_id and newreq.equipment_id:
+            newreq.project_id = newreq.equipment_id.project_id
+        if not newreq.task_id and newreq.maintenance_type == "preventive":
+            newreq.task_id = newreq.equipment_id.preventive_default_task_id
+
+        return newreq
+
+    @api.onchange("equipment_id")
+    def onchange_equipment_id(self):
+        super(MaintenanceRequest, self).onchange_equipment_id()
```
Comment (victoralmau) [inline maintenance_project/models/maintenance_request.py]:
```suggestion
        res = super().onchange_equipment_id()
```

File: maintenance_project/models/maintenance_request.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        if self.equipment_id and self.equipment_id.project_id:
+            self.project_id = self.equipment_id.project_id
+        return
```
Comment (victoralmau) [inline maintenance_project/models/maintenance_request.py]:
```suggestion
        return res
```

File: maintenance_project/models/project_project.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    def _compute_equipment_count(self):
+        for project in self:
+            project.equipment_count = len(project.equipment_ids)
+
+    def action_view_equipment_ids(self):
+        """
+        Access to the current equipments for this project
+        """
+        self.ensure_one()
+        action = self.env.ref("maintenance.hr_equipment_action").read()[0]
```
Comment (victoralmau) [inline maintenance_project/models/project_project.py]:
```suggestion
        action = self.env["ir.actions.actions"]._for_xml_id(
            "maintenance.hr_equipment_action"
        )
```

File: maintenance_project/models/project_project.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            project.maintenance_request_count = len(
+                project.maintenance_request_ids.filtered(lambda x: not x.stage_id.done)
+            )
+
+    def action_view_maintenance_request_ids(self):
+        """
+        Access to the undone maintenance requests for this project
+        """
+        self.ensure_one()
+        action = self.env.ref("maintenance.hr_equipment_request_action").read()[0]
```
Comment (victoralmau) [inline maintenance_project/models/project_project.py]:
```suggestion
                action = self.env["ir.actions.actions"]._for_xml_id(
            "maintenance.hr_equipment_request_action"
        )
```

File: maintenance_project/tests/test_maintenance_project.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        super().setUp()
+
+        self.cron = self.env.ref("maintenance.maintenance_requests_cron")
+        self.project1 = self.env["project.project"].create({"name": "My project"})
+        self.project_demo1 = self.env["project.project"].create(
+            {
+                "name": "General equipment project",
+                "user_id": self.env.ref("base.user_admin").id,
+            }
+        )
```
Comment (victoralmau) [inline maintenance_project/tests/test_maintenance_project.py]:
```suggestion
        self.project_demo1 = self.env.ref("maintenance_project.project_project_1")
```

File: maintenance_project/tests/test_maintenance_project.py [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+        self.equipment_prj1_demo3 = self.env["maintenance.equipment"].create(
+            {
+                "name": self.project_demo1.name,
+                "create_project_from_equipment": True,
+                "project_id": self.project_demo1.id,
+            }
+        )
+        self.project_demo1._compute_equipment_count()
```
Comment (victoralmau) [inline maintenance_project/tests/test_maintenance_project.py]:
```suggestion
        self.equipment_demo1 = self.env.ref("maintenance_project.equipment_1")
        self.equipment_demo2 = self.env.ref("maintenance_project.equipment_2")
        self.equipment_demo3 = self.env.ref("maintenance_project.equipment_3")
```

**Review by victoralmau — CHANGES_REQUESTED** [review]
The module cannot even be installed (changes applied in tests are not necessary).

**Comment by mymage** [comment]
@victoralmau thank you very much for your help

**Review by victoralmau — APPROVED** [review]

**Pull request merged**